Added necessary changes in create record function

AnastasiaShemyakinskaya opened the pull request:
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com

AnastasiaShemyakinskaya committed:
Added necessary changes in create record function
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
   private final AtomicInteger rrc = new AtomicInteger(0);
   private final Map<String, Properties> configCache = new ConcurrentHashMap<>();
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
   private final AtomicInteger rrc = new AtomicInteger(0);
   private final Map<String, Properties> configCache = new ConcurrentHashMap<>();
   private final Map<Properties, AdminClientCreateFunctionImpl> adminClientCreateFuncCache =
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
   private final Map<Properties, AdminClientCreateFunctionImpl> adminClientCreateFuncCache =
-    new ConcurrentHashMap<>();
+      new ConcurrentHashMap<>();
   private final Map<String, AdminClient> adminClientCache = new ConcurrentHashMap<>();
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
-    new ConcurrentHashMap<>();
+      new ConcurrentHashMap<>();
   private final Map<String, AdminClient> adminClientCache = new ConcurrentHashMap<>();
   private final Map<Properties, ProducerCreateFunctionImpl> producerCreateFuncCache =
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
   private final Map<Properties, ProducerCreateFunctionImpl> producerCreateFuncCache =
-    new ConcurrentHashMap<>();
+      new ConcurrentHashMap<>();
   private final Map<String, KafkaProducer> producerCache = new ConcurrentHashMap<>();
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
-    new ConcurrentHashMap<>();
+      new ConcurrentHashMap<>();
   private final Map<String, KafkaProducer> producerCache = new ConcurrentHashMap<>();
   private final Map<AdminClient, TopicCreateFunctionImpl> topicCreateFuncCache =
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -46,23 +45,23 @@
   private final Map<AdminClient, TopicCreateFunctionImpl> topicCreateFuncCache =
-    new ConcurrentHashMap<>();
+      new ConcurrentHashMap<>();
   private final Map<String, NewTopic> topicCache = new ConcurrentHashMap<>();
```
needs to be cleaned up in the `doClose` method

akurilov commented on src/main/java/com/emc/mongoose/storage/driver/kafka/KafkaStorageDriver.java:
```diff
@@ -150,60 +149,64 @@ private Properties createConfig(String nodeAddr) {
       val producerConfig =
-        producerCreateFuncCache.computeIfAbsent(config, ProducerCreateFunctionImpl::new);
+          producerCreateFuncCache.computeIfAbsent(config, ProducerCreateFunctionImpl::new);
       val kafkaProducer = producerCache.computeIfAbsent(nodeAddr, producerConfig);
```
Is the `kafkaProducer` thread safe? The cached pool should be used if not. See the example in the pravega storage driver @ master branch.

akurilov replied:
kafkaProducer is thread safe
however, probably we still should use the pool with the size = `ioWorkerCount` for the multithreaded producing which may be more efficient...

AnastasiaShemyakinskaya committed:
Added doClose method
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com

AnastasiaShemyakinskaya committed:
Fixed ci
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my knowledge, is covered under an appropriate open source license and I have the right under that license to submit that work with modifications, whether created in whole or in part by me, under the same open source license (unless I am permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and that a record of the contribution (including all personal information I submit with it, including my sign-off) is maintained indefinitely and may be redistributed consistent with this project or the open source license(s) involved.

Signed-off-by: Anastasia Shemyakinskaya shem98a@gmail.com
